### DISCUSSION TITLE
Safely move Flux components (switching branch)

### DISCUSSION THREAD

**Discussion by Thaval** [fluxcd/flux2#2711] [Q&A] [answered]
Hi,

imagine I've setup flux components in a wrong branch like 'feature/test' and flux already manages the lifecycle of the k8s objects. Now I want to use another git branch as my source. How can I do that without breaking/deleting anything? 
The deployed state should not be changed, I only want to change the source (well, if the source is different, then it's fine if flux runs "kubectl apply..."). 

Kind regards

**Comment by stefanprodan** [comment, marked as answer]
Assuming the new branch has the same structure and manifests, you can rerun `flux bootstrap` for that branch. Another option is to use `flux uninstall --keep-namespace` (this will remove Flux but will leave all workloads in place), then reinstall for the new branch.

**Comment by Thaval** [comment]
Nice one, thanks. I'll try it out tomorrow. I definitely want to use `--keep-namespace` as I just want to move, not really delete my workload.

**Reply by stefanprodan** [reply]
`flux uninstall` does not remove your workloads only the Flux controllers and CRDs. The `--keep-namespace` leaves the `flux-system` namespace in place to preserve the SSH deploy key and any other controllers you may have deployed to flux-system ns. It is advised to no deploy anything in the flux-system besides Flux custom resources.

**Reply by Thaval** [reply]
Yea currently I store the flux CRDs (HelmRelease, GitRepository, etc.) in the flux-system namespace and my workloads are stored in different namespace than flux-system.
